parser-typechecker/src/Unison/UnisonFile/Names.hs

```diff
@@ -31,8 +26,8 @@ import qualified Unison.WatchKind as WK
 toNames :: Var v => UnisonFile v a -> Names
 toNames uf = datas <> effects
   where
-    datas = foldMap DD.Names.dataDeclToNames' (Map.toList (UF.dataDeclarationsId uf))
-    effects = foldMap DD.Names.effectDeclToNames' (Map.toList (UF.effectDeclarationsId uf))
+    datas = foldMap (DD.Names.dataDeclToNames' Name.unsafeFromVar) (Map.toList (UF.dataDeclarationsId uf))
```

Comment:
Just curious, what would be involved in adding a non-empty guarantee to Vars so that this isn't unsafe?

Reply:
I think the `Var` <-> `Name` conversions need a little (or a lot) more elbow grease before we can have a safe conversion. Right now I think it's only really sensible to convert a `Var` into a `Name` if it happens to have `User` type, which was actually typed in by a user (and so looks something like `.foo.bar.baz`). But we have a couple places where we stuff non-user-written things into the `User` type (like `User "()"`, for foralls).